netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -111,9 +116,22 @@ public NettyClientHandler(BufferingHttp2ConnectionEncoder encoder, Http2Connecti
     nextStreamId = connection.local().nextStreamId();
     connection.addListener(new Http2ConnectionAdapter() {
       @Override
+      public void onStreamClosed(Http2Stream stream) {
+        // Report the closure status for the stream.
+        reportStatus(clientStream(stream), null, false);
+      }
+
+      @Override
       public void onGoAwayReceived(int lastStreamId, long errorCode, ByteBuf debugData) {
         goAwayStatus(statusFromGoAway(errorCode, debugData));
```

Comment:
Are we sure this is called before `onStreamClosed()` for any streams that will be closed due to the GOAWAY?